File: src/test/baekjoon/python/twopointer/SumOfNumbers2.py
N, M = map(int, input().split())
arr = list(map(int, input().split()))
start, end = 0, 0

answer = 0
while end <= N and start <= end:
    sum_of_nums = sum(arr[start:end])

    if sum_of_nums == M:
        answer += 1
        start += 1
    elif sum_of_nums > M:
        start += 1
    else:
        end += 1

print(answer)

File: src/test/baekjoon/java/twopointer/SumOfNumbers2.java
package test.baekjoon.java.twopointer;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;

public class SumOfNumbers2 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());

        st = new StringTokenizer(br.readLine());
        List<Integer> arr = new ArrayList<>();
        for(int i = 0; i < N; i++) {
            arr.add(Integer.parseInt(st.nextToken()));
        }

        int start = 0, end = 0, answer = 0;

        while(end <= N && start <= end) {
            int sum = arr.subList(start, end).stream().mapToInt(i -> i).sum();

            if(sum == M) {
                answer++;
                start++;
            } else if(sum > M) {
                start++;
            } else {
                end++;
            }
        }

        System.out.println(answer);
    }
}
